src/components/cart/CheckRecipe.tsx

```diff
@@ -39,58 +38,75 @@ const CheckRecipe: FC<Props> = ({ recipeInCart }) => {
       </Text>
 
       {Object.keys(ingredients.productsByStoreId).map((storeId) =>
-        ingredients.productsByStoreId[storeId].map((product) => (
-          <Flex
-            p={"4px 32px 4px 48px"}
-            alignItems={"center"}
-            justifyContent={"space-between"}
-            minH={"80px"}
-            bg={"white"}
-            mb={"1px"}
-            key={product.id}
-          >
-            <Flex flexDir={"row"} w={"720px"} alignItems={"center"}>
-              <Image width={"64px"} src={product.image} />
-              <Text
-                color={"rgb(28, 37, 41)"}
-                fontSize={" 14px"}
-                lineHeight={"1.47"}
-                pl={"48px"}
-              >
-                {product.name}
-              </Text>
-              <Text pl={"4"}>{product.textualAmount}</Text>
+        ingredients.productsByStoreId[storeId].map((product) => {
+          const amount =
+            ingredientsInCart.find(
+              (ingredientInCart) => ingredientInCart.id === product.id,
+            )?.amount || 0;
+          const amountInCart =
+            ingredientsInCart.find(
+              (ingredientInCart) => ingredientInCart.id === product.id,
+            )?.amountInCart || 0;
```

Comment:
Pojdme se o tomhle pobavit, takhle logika by mela byt mimo render. Musime nejak znovu promyslet jak pracujeme s tim globalnim stavem, protoze mi prijde ze mame strasne moc logiky navazanou na render, coz je skoda, protoze to pak zeslozituje praci s UI.